build-tools/packages/build-cli/src/commands/generate/layerGeneration.ts

```diff
@@ -0,0 +1,224 @@
+ * Licensed under the MIT License.
+ */
+
+import { existsSync } from "node:fs";
+import { readFile, writeFile } from "node:fs/promises";
+import path from "node:path";
+import type { Logger, Package } from "@fluidframework/build-tools";
+import { Flags } from "@oclif/core";
+import { diff, parse } from "semver";
+import { PackageCommand } from "../../BasePackageCommand.js";
+import type { PackageSelectionDefault } from "../../flags.js";
+
+// Approximate month as 33 days to add some buffer and avoid over-counting months in longer spans.
+export const daysInMonthApproximation = 33;
+
+export default class UpdateGenerationCommand extends PackageCommand<
+	typeof UpdateGenerationCommand
+> {
+	static readonly description =
+		"Updates the generation and release date for layer compatibility.";
+
+	static readonly flags = {
+		generationDir: Flags.directory({
+			description: "The directory where the generation file is located.",
+			default: "./src",
+		}),
+		generationFileName: Flags.string({
+			description: `The file name for the generation file.`,
+			default: `layerGenerationState.ts`,
+		}),
+		minimumCompatWindowMonths: Flags.integer({
+			description: `The minimum compatibility window in months that is supported across all Fluid layers.`,
+			default: 3,
+		}),
+		...PackageCommand.flags,
+	} as const;
+
+	protected defaultSelection = "dir" as PackageSelectionDefault;
+
+	protected async processPackage(pkg: Package): Promise<void> {
+		const { generationDir, generationFileName, minimumCompatWindowMonths } = this.flags;
+		const generationFile = path.join(pkg.directory, generationDir, generationFileName);
+
+		if (!existsSync(generationFile)) {
+			// Early exit; no error.
+			this.info(`File ${generationFile} doesn't exist; skipping.`);
+			return;
+		}
+
+		const currentPkgVersion = pkg.version;
+		const generationFileContents = await readFile(generationFile, "utf8");
+		const newGeneration = maybeGetNewGeneration(
+			generationFileName,
+			generationFileContents,
+			minimumCompatWindowMonths,
+			currentPkgVersion,
+			this.logger,
+		);
+		if (newGeneration === undefined) {
+			// No update needed; early exit.
+			return;
+		}
+
+		const releaseDateFormatted = formatDateForLayerFile(new Date());
+		await writeFile(
+			generationFile,
+			generateLayerFileContent(newGeneration, releaseDateFormatted, currentPkgVersion),
+			{
+				encoding: "utf8",
+			},
+		);
+		this.info(`Layer generation updated to ${newGeneration}`);
+	}
+}
+
+/**
+ * Formats a Date object into the MM/DD/YYYY string format used in layer generation files.
+ *
+ * @param date - The Date object to format
+ * @returns A string in MM/DD/YYYY format with zero-padded month and day values
+ *
+ * @example
+ * ```typescript
+ * const date = new Date(2025, 10, 29); // November 29, 2025
+ * const formatted = formatDateForLayerFile(date);
+ * console.log(formatted); // "11/29/2025"
+ * ```
+ */
+export function formatDateForLayerFile(date: Date): string {
```

Comment:
I think we have date-fns available, and I strongly recommend using it rather than implementing. Though I guess this is simple enough as long as you have good unit test coverage.